### PR TITLE
docs: Custom Cypress test doc improvements

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to improve the `Custom Cypress test` docs.

### CODE REVIEW

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 1]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
 ![checks-integration-tests](https://vtexhelp.vtexassets.com/assets/docs/src/custom-test-checks___583ac9b78e43ebdafde93a735a47d61b.png)
 
 6. Review the results to check the execution status of your custom integration tests.
+
+---
+
+## Exclusive execution of your custom Cypress tests
+
+With continuous development, some storefront projects may have changes in the components and/or pages's code that cause errors in the out of the box end-to-end tests that are run for all FastStore projects.
```
Comment (hellofanny) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:72]: Due to ongoing development, some storefront projects may have code changes in their components and/or pages. These changes can lead to errors in the out-of-the-box end-to-end tests that run for all FastStore projects.

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 2]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+
+describe('Custom PLP tests', () => {
+  it('visit collection page', () => {
+    cy.visit(pages.collection)
+    cy.get('[data-testid="product-gallery"]').should('exist').and('be.visible')
+  })
+})
+```
+
+Some projects may prefer exclusive execution of its custom Cypress tests. Although it is not our recommendation, for projects using Cypress version 12.x or higher there is the possibility of running the **Integration Tests** check only for your custom tests.
```
Comment (hellofanny) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:89]:
```suggestion
Although we recommend keeping the default Cypress tests, it is possible to run the Integration Tests check exclusively for your custom tests for projects using Cypress version 12.x or higher."
```

or 

Although we do not recommend exclusive execution of custom Cypress tests for some projects, it is possible to run the Integration Tests check exclusively for your custom tests for projects using Cypress version 12.x or higher.

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 3]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+
+For that, first you need to use at least Cypress 12 by setting the `cypressVersion` experimental feature flag:
```
Comment (hellofanny) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:91]:
```suggestion
For that, you must first ensure that you are using Cypress version 12 or higher by setting the `cypressVersion` experimental feature flag:
```

Comment (hellofanny) [reply apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:91]: should we reinforce that so far we only support till cypress 12?

Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:91]:
```suggestion
Follow the steps below to configure your project to only run custom Cypress tests:
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 4]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:92]:
```suggestion
## Step by step

### Step 1 - Setting Cypress version

First, ensure that you use Cypress version 12 or higher by setting the `cypressVersion` experimental feature flag:
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 5]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+3. Set the `cypressVersion` value to `12`.
+
+```js filename="faststore.config.js" {4}
+  },
+  experimental: {
+      enableCypressExtension: true,
+      cypressVersion: 12,
+  }
+```
+
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:104]:
```suggestion
### Step 2 - Updating the Cypress configuration file
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 6]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+Furthermore, you need to update your Cypress configuration file (`cypress.json`) to match the latest Cypress configuration file format. Refer to the official [Cypress configuration file](https://docs.cypress.io/guides/references/configuration) guide.
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:105]:
```suggestion
Ensure that your Cypress configuration file (cypress.json) matches the latest Cypress configuration file format. To do so, refer to the official [Cypress configuration file](https://docs.cypress.io/guides/references/configuration) guide.
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 7]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:106]:
```suggestion
### Step 3 - Changing the custom Cypress tests directory
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 8]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+Finally, you need to change your custom Cypress tests directory from `cypress/integration` to `cypress/e2e`:
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:107]:
```suggestion
Now, change your custom Cypress tests directory from `cypress/integration` to `cypress/e2e`:
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 9]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+
+1. In your storefront project, rename the folder from `cypress/integration` to `cypress/e2e`.
+2. Also change your new Cypress configuration file `e2e` object to match the new custom tests folder `cypress/e2e`:
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:110]:
```suggestion
2. Update your new Cypress configuration file `cypress.config.ts` to match the new custom tests folder `cypress/e2e` by specifying the `specPattern`:
```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 10]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+  ...,
+  e2e: {
+    ...,
+    specPattern: 'cypress/e2e/**/*.test.{js,jsx,ts,tsx}',
+  },
+})
+```
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:122]:
```suggestion
```
3. Run `yarn dev` in the terminal to sync the changes you have made.

```

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx (hunk 2 above)
Comment (Mariana-Caetano) [reply apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:89]: @hellofanny and @lucasfp13, I combined our suggestions and added a callout suggestion above :) let me know what you think.

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 11]
```diff
@@ -64,3 +64,59 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+const { pages } = cypress
+
+describe('Custom PLP tests', () => {
+  it('visit collection page', () => {
+    cy.visit(pages.collection)
+    cy.get('[data-testid="product-gallery"]').should('exist').and('be.visible')
+  })
+})
+```
+
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:88]: > Although exclusive execution of custom tests is possible for projects using Cypress version 12.x or higher, we do not recommend this approach. If possible, keep the default Cypress tests.

Maybe? 🙈

[the remark this answers is not in the document]
Comment (Mariana-Caetano) [reply apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:88]: Yes! 🚀

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx (hunk 3 above)
Comment (lucasfp13) [reply apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:91]: I think we can remove the `or higher` expression and specifying only the version 12

**Comment by lucasfp13** [comment]
> When we use "for some projects", why is it not recommended for some projects? 👀

We do not recommend this for **any** FastStore project because they will be skipping the OOTB integration tests

**Review by hellofanny — APPROVED** [review]
🚀

File: apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx [hunk 12]
```diff
@@ -64,3 +64,75 @@ Like any other project using Cypress, you'll follow standard practices, includin
...
+      enableCypressExtension: true,
+      cypressVersion: 12,
+  }
+```
+
+### Step 2 - Updating the Cypress configuration file
+
+Ensure that your Cypress configuration file (`cypress.json`) matches the latest Cypress configuration file format. To do so, refer to the official [Cypress configuration file](https://docs.cypress.io/guides/references/configuration) guide.
+
+### Step 3 - Changing the custom Cypress tests directory
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/faststore-config-options/custom-cypress-test.mdx:119]:
```suggestion
#### Step 3 - Changing the custom Cypress tests directory
```

**Review by Mariana-Caetano — APPROVED** [review]
I left some suggestions about the headings, but I've already pre-approved 🚀

**Pull request merged**